src/apis/auth/endpoint.ts

```diff
@@ -0,0 +1,37 @@
+import { BeforeAuthInstance } from '../axiosInstance';
+import {
+  LoginRequest,
+  LoginReturn,
+  RegisterRequest,
+  RegisterReturn,
+} from './type';
+
+//로그인 api
+export async function postToken(loginInfo: LoginRequest): Promise<LoginReturn> {
+  //email, password encoding
+  const combined = `${loginInfo.email}:${loginInfo.password}`;
+  const encodeCombined = btoa(combined);
+  const response = await BeforeAuthInstance.post(
+    '/tokens',
+    {},
+    {
+      headers: {
+        Authorization: `Basic ${encodeCombined}`,
+      },
+    }
+  );
+  return response.data;
+}
+
+//회원가입 api
+export async function postUserRegister(
+  registerInfo: RegisterRequest
+): Promise<RegisterReturn> {
+  const response = await BeforeAuthInstance.post('/users', {
+    email: registerInfo.email,
+    password: registerInfo.password,
+    nickname: registerInfo.nickname,
+    mbti: null,
+  });
+  return response.data;
+}
```

Comment:
```suggestion
export async function postUserRegister(
  registerInfo: RegisterRequest
): Promise<RegisterReturn> {
  const response = await BeforeAuthInstance.post('/users', { params: registerInfo });
  return response.data;
}
```
body로 보내는 데이터들을 하나하나 나열해줄 필요 없이 한번에 보낼 수 있습니다!

Reply:
@rhehfl 이건 몰랐는데 완전 좋네요ㅎㅎ 감사합니다!

Reply:
@rhehfl 방금 로직을 적용시키고 테스트를 해봤는데, `params`를 사용하면 쿼리파라미터로 값이 들어가서 `body`에 넣는 값과는 다른거 아닌가요?🤔

Reply:
헉 😓
```ts
const response = await BeforeAuthInstance.post('/users',  registerInfo);
```
입니다 죄송합니다 혼란을 드린 것 같네요

Reply:
@rhehfl 아하! 이해했습니다! 감사합니다!